.github/CODEOWNERS

```diff
@@ -0,0 +1,3 @@
+* @aws-observability/adot-collector-maintainers @aws-observability/aws-application-signals-maintainers
+
+sample-apps/prometheus-sample-app     @aws-observability/adot-collector-maintainers
```

Comment:
I thought `@aws-observability/aws-application-signals-maintainers` would only own `sample-apps` and `@aws-observability/adot-collector-maintainers` would own everything else plus `sample-apps/prometheus-sample-app`, no?

Expected something like
```
* @aws-observability/adot-collector-maintainers

sample-apps @aws-observability/aws-application-signals-maintainers
sample-apps/prometheus-sample-app @aws-observability/adot-collector-maintainers
```